### DISCUSSION TITLE
Codelist uploaded to OpenCodelists missing a column

### DISCUSSION THREAD

**Discussion by SophieEastwood** [opensafely/documentation#442] [Codelists] [answered]
I have just uploaded a diabetes codelist onto opencodelists. The main reason for uploading it (as there are already several diabetes codelists on the platform) was to attach categories of diagnosis to each code (i.e. a 3rd column in the spreadsheet, after the CVTV3 code and the term). But the categories column does not seem to have uploaded, only the first 2 columns.
 
Can you help please? There seem to be other codelists from external sources with categories attached. Thanks in advance!

**Comment by rohinimathur** [comment]
Hi @SophieEastwood could you please post your solution here and rename the discussion so we can incorporate this into the standard documentation?

**Reply by SophieEastwood** [reply]
Hi @rohinimathur sorry I did not find a technical solution I'm afraid. Decided to extract by category then amalgamate, rather than labelling the categories at extraction. But there are some codelists on Opencodelists (e.g. Exeter diabetes one) which do have categories attached, so it must be possible!

**Comment by inglesp** [comment, marked as answer]
Hello, sorry for the slow response.

There's an undocumented backdoor that allows you to upload codelists with categories.  You need to go to https://www.opencodelists.org/codelist/user/YOUR_USERNAME/add/.  Note that unlike the normal, the CSV you upload must have headers, and the upload won't be validated in any way.

Also, if you've tried to upload a codelist with categories via the normal route, you've probably ended up with a codelist you'd like to delete.  There's currently no way to do this in the web interface, but if you let me know the URL of a codelist you'd like to delete, I can delete it for you.

Hope this helps!

**Reply by SophieEastwood** [reply]
Hi Peter

Thanks for the reply - useful to know!

Great, yes could you delete this please:
https://www.opencodelists.org/codelist/user/Sophie_Eastwood/diabetes_diagnostic__nondiagnostic_categories_ctv3_codes/57874bfc/

Thanks a lot
Sophie
